Google_test.py

```diff
@@ -3,22 +3,108 @@
 
 import unittest
 from Google import Google
+from Googler import Googler
 
-class TestDistance(unittest.TestCase):
-    #TODO: Write test cases for distance function
-    return
+class TestSearch(unittest.TestCase):
 
-class TestUsageSimilarity(unittest.TestCase):
-    #TODO: Write test cases for usage similarity function
-    return
+    def setUp(self):
+        self.googlers = [Googler({  "id": 0, "department": 10, "cost_center":10, "manager_id":-1,
+                                    "location": 10, "lowest_dir_id":10, "job_family": 10}),
+                        Googler({   "id": 14, "department": 10, "cost_center":10, "manager_id":2,
+                                    "location": 10, "lowest_dir_id":10, "job_family": 10})]
+        self.google = Google(self.googlers)
+        self.actual_googler = Googler({ "id": 14, "department": 10, "cost_center":10, "manager_id":2,
+                                        "location": 10, "lowest_dir_id":10, "job_family": 10})
+
+    def test_finds_googler(self):
+        #Test that the correct googler is found
+        self.assertEqual(self.actual_googler, self.google.search(14))
+
+    def test_raises_error(self):
+        #Test that an error is thrown when searching for Googler that doesn't exist
+        self.assertRaises(ReferenceError, self.google.search, 1000)
 
-class TestSearch(unittest.TestCase):
-    #TODO: Write test cases for search function
-    return
 
 class TestManagerHierarchy(unittest.TestCase):
-    #TODO: Write test cases to ensure manager hierarchy tree is implemented correctly
-    return
+
+    def setUp(self):
+        self.googlers = [Googler({  "id": 0, "department": 10, "cost_center":10, "manager_id":-1,
+                                    "location": 10, "lowest_dir_id":10, "job_family": 10}),
+                        Googler({   "id": 14, "department": 10, "cost_center":10, "manager_id":0,
+                                    "location": 10, "lowest_dir_id":10, "job_family": 10})]
+        self.google = Google(self.googlers)
+        self.actual_googler = Googler({ "id": 14, "department": 10, "cost_center":10, "manager_id":0,
+                                        "location": 10, "lowest_dir_id":10, "job_family": 10})
+        self.actual_manager = Googler({ "id": 0, "department": 10, "cost_center":10, "manager_id":-1,
+                                        "location": 10, "lowest_dir_id":10, "job_family": 10})
+
+    def test_get_manager(self):
+        # Tests that a googles manager is retrieved
+        googler = Googler({ "id": 14, "department": 10, "cost_center":10, "manager_id":0,
+                                        "location": 10, "lowest_dir_id":10, "job_family": 10})
+        manager = self.google.company.get(googler.manager_id)
+        self.assertEqual(self.actual_manager, manager)
+
+    def test_get_ceo(self):
+        # Test that the ceo has a None manager object
+        self.assertEqual(None, None)
```

Comment:
Going to implement this test case when we receive updated dataset and I know how user id to manager id relationship will work.